qcelemental/molparse/to_string.py

```diff
@@ -143,6 +144,31 @@ def to_string(molrec: Dict,
             'coordinates': 'cartesian',
         }
 
+    elif dtype == 'molpro':
+
+        atom_format = '{elem}'
+        ghost_format = '{elem}'
+        umap = {'bohr': 'bohr', 'angstrom': 'angstrom'}
+
+        atoms = _atoms_formatter(molrec, geom, atom_format, ghost_format, width, prec, 2)
+
+        ghost_atoms = []
+        nat = geom.shape[0]
+        for iat in range(nat):
+            if not molrec['real'][iat]:
+                ghost_atoms.append(str(iat+1))
+
+        first_line = f"""{{{umap.get(units.lower())}}}"""
+        second_line = """geometry={"""
+        end_bracket = """}"""
+        smol = [first_line, second_line]
+        smol.extend(atoms)
+        smol.append(end_bracket)
+
+        if len(ghost_atoms) != 0:
+            ghost_line = f"""dummy,{",".join(ghost_atoms)}"""
+            smol.append(ghost_line)
```

Comment:
can be a little more compact

```
reals = [True, False, True, True, False]  # molrec['real']

ans = 'dummy,' + ','.join([str(idx + 1) for idx, real in enumerate(reals) if not real])

print(ans)
# dummy,2,5
```

Reply:
I've updated the code to make it more compact